rct229/rules/section5/section5rule13.py

```diff
@@ -0,0 +1,126 @@
+    RuleDefinitionListIndexedBase,
+)
+from rct229.rule_engine.user_baseline_proposed_vals import UserBaselineProposedVals
+from rct229.ruleset_functions.get_opaque_surface_type import (
+    OpaqueSurfaceType as OST,
+    get_opaque_surface_type,
+)
+from rct229.ruleset_functions.get_surface_conditioning_category_dict import (
+    SurfaceConditioningCategory as SCC,
+    get_surface_conditioning_category_dict,
+)
+from rct229.utils.jsonpath_utils import find_all
+from rct229.utils.std_comparisons import std_equal
+
+
+class Section5Rule13(RuleDefinitionListIndexedBase):
+    """Rule 5 of ASHRAE 90.1-2019 Appendix G Section 5 (Envelope)"""
+
+    def __init__(self):
+        super(Section5Rule13, self).__init__(
+            rmrs_used=UserBaselineProposedVals(False, True, False),
+            required_fields={
+                "$": ["weather"],
+                "weather": ["climate_zone"],
+            },
+            each_rule=Section5Rule13.BuildingRule(),
+            index_rmr="baseline",
+            id="5-13",
+            description="Baseline floor assemblies must  match the appropriate assembly maximum U-factors in Tables G3.4-1 through G3.4-9.",
+            list_path="ruleset_model_instances[0].buildings[*]",
+        )
+
+    def create_data(self, context, data=None):
+        rmr_baseline = context.baseline
+        return {"climate_zone": rmr_baseline["weather"]["climate_zone"]}
+
+    class BuildingRule(RuleDefinitionListIndexedBase):
+        def __init__(self):
+            super(Section5Rule13.BuildingRule, self).__init__(
+                rmrs_used=UserBaselineProposedVals(False, True, False),
+                required_fields={"$..zones[*]": ["surfaces"]},
+                each_rule=Section5Rule13.BuildingRule.FloorRule(),
+                index_rmr="baseline",
+            )
+
+        def create_context_list(self, context, data=None):
+            building = context.baseline
+            # List of all baseline floor surfaces to become the context for RoofRule
+            return [
+                UserBaselineProposedVals(None, surface, None)
+                for surface in find_all("$..surfaces[*]", building)
+                if get_opaque_surface_type(surface) == OST.FLOOR
+            ]
+
+        def create_data(self, context, data=None):
+            building = context.baseline
+            # Merge into the existing data dict
+            return {
+                **data,
+                "surface_conditioning_category_dict": get_surface_conditioning_category_dict(
+                    data["climate_zone"], building
+                ),
+            }
+
+        class FloorRule(RuleDefinitionBase):
+
+            def __init__(self):
+                super(Section5Rule13.BuildingRule.FloorRule, self).__init__(
+                    rmrs_used=UserBaselineProposedVals(False, True, False),
+                    required_fields={"$": ["construction"],
+                                     "construction": ["u_factor"]},
```

Comment:
We are skipping Black in the git PR for now, but we should still use it before committing. This looks like non-Black formatting.